### PR TITLE
#295 FbStatus.Check replaced by org.takes.misc.Condition

### PR DESCRIPTION
**Pull request by komaz**
Fixed issue #295, also added test cases for FbStatus

### CODE REVIEW

**Comment by davvd** [comment]
@komaz Let me find a reviewer for this pull request, thanks for submitting it

**Comment by davvd** [comment]
@pinaf please review,thanks

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+/**
+ * Test case for {@link FbStatus}.
+ * @author Ivan Inozemtsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.16.10
+ */
+public final class FbStatusTest {
+
+    /**
+     * FbStatus should react on status equal to passed in constructor.
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz Let's change this to `FBStatus can react to correct status`

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void reactsExactStatus() throws IOException {
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz rename to `reactsCorrectStatus()` for consistency

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                    status,
+                    new TkFixed(new RsText("not found response"))
+                ).route(req).next()
+            ).printBody(),
+            Matchers.startsWith("not found")
+        );
+    }
+
+     /**
+     * FbStatus should react on status which fits to condition passed in
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz `FBStatus can react to Condition`

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+     * constructor.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void reactsFittingStatus() throws IOException {
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz `reactsToCondition()`

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+ * Test case for {@link FbStatus}.
+ * @author Ivan Inozemtsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.16.10
+ */
+public final class FbStatusTest {
+
+    /**
+     * FbStatus should react on status equal to passed in constructor.
+     * @throws IOException If some problem inside
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz Let's go with the general type `Exception` in all unit tests

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                ).route(req).next()
+            ).printBody(),
+            Matchers.startsWith("not found")
+        );
+    }
+
+     /**
+     * FbStatus should react on status which fits to condition passed in
+     * constructor.
+     * @throws IOException If some problem inside
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz Let's go with the general type `Exception` in all unit tests

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                    new FbFixed(new RsText("response text"))
+                ).route(req).next()
+            ).printBody(),
+            Matchers.startsWith("response")
+        );
+    }
+
+    /**
+     * FbStatus should react on status equal to passed in constructor.
+     * @throws IOException If some problem inside
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz Let's go with the general type `Exception` in all unit tests

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                    },
+                    new FbFixed(new RsText("response text"))
+                ).route(req).next()
+            ).printBody(),
+            Matchers.startsWith("response")
+        );
+    }
+
+    /**
+     * FbStatus should react on status equal to passed in constructor.
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz `FbStatus can ignore different status`

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void ignoresOtherStatus() throws IOException {
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz `ignoresDifferentStatus()`

**Comment by pinaf** [comment]
@komaz looks good. 9 minor comments above.

**Comment by komaz** [comment]
@pinaf thanks! Just pushed suggested changes

File: src/test/java/org/takes/facets/fallback/FbStatusTest.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                    new TkFixed(new RsText("not found response"))
+                ).route(req).next()
+            ).printBody(),
+            Matchers.startsWith("not found")
+        );
+    }
+
+     /**
+     * FbStatus can react to Condition.
+     *
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fallback/FbStatusTest.java]: @komaz let's remove this blank line

**Comment by pinaf** [comment]
@komaz thanks. 1 more and we are good to go!

**Comment by komaz** [comment]
@pinaf no problem, done. Just a side question, do you think it's a good practice to squash all commits with styling fixes into a singe one? As until it's merged I can safely rewrite a history

**Comment by pinaf** [comment]
@komaz no need to squash. however, you removed the wrong blank lines. I asked to remove a single line in the javadoc

**Comment by komaz** [comment]
@pinaf sorry, my bad

**Comment by pinaf** [comment]
@rultor merge
@komaz thanks!

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka please, review this task for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by ypshenychka** [comment]
@komaz 
According to our [QA Rules](http://www.teamed.io/qa.html):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

Please expand description to PR with appropriate information,

**Comment by komaz** [comment]
@ypshenychka please take a look at updated description

**Comment by ypshenychka** [comment]
@komaz Great! Thanks

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@pinaf just added 10 mins to @ypshenychka (for QA), payment ID is `57688942`

**19 mins** sent to your balance (ID `AP-1KD21491BF8407901`), many thanks! It took 9 hours and 11 mins.

extra minutes for review comments (c=4)

+19 to your rating, your total score is [+8095](http://www.netbout.com/b/35128?open=rating)

**Comment by pinaf** [comment]
@yegor256 @ypshenychka why c=4 if I made 9 remarks?

**Comment by yegor256** [comment]
@pinaf we count only issue messages now, not code remarks

**Comment by pinaf** [comment]
@yegor256 ?? So if I find more problems with the code I don't get a bonus but if I say anything in the issue I do? I don't understand the logic - am I missing something?

**Comment by yegor256** [comment]
@pinaf there is no logic, it's just a bug in our measurement software. I'm working on this. For now, that's the way it is.

**Comment by pinaf** [comment]
@yegor256 ok, thanks for the explanation.

**Pull request merged**